keyboards/xd75/keymaps/tdl-jturner/keymap.c

```diff
@@ -0,0 +1,304 @@
+    {KC_LYDEF,________,________,________,________,________,___XX___,___XX___,___XX___,___XX___,___XX___,KC_MS_U ,KC_WH_U ,KC_ACL2 ,________},
+    {________,________,________,________,________,________,___XX___,___XX___,___XX___,___XX___,KC_MS_L ,KC_ACL0 ,KC_MS_R ,KC_ACL1 ,___XX___},
+    {________,________,________,________,________,________,___XX___,___XX___,___XX___,___XX___,___XX___,KC_MS_D ,KC_WH_D ,KC_BTN2 ,___XX___},
+    {________,________,________,________,________,________,________,________,________,________,KC_BTN1 ,KC_BTN3 ,KC_BTN4 ,___XX___,___XX___}
+  },
+  //Systen Keys
+  [_LYSYS] =
+   {
+     {________,________,________,________,________,________,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___},
+     {KC_LYDEF,________,________,________,________,________,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,________},
+     {________,________,________,________,________,________,___XX___,___XX___,___XX___,___XX___,___XX___,QWERTY  ,COLEMAK ,___XX___,RESET   },
+     {________,________,________,________,________,________,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___},
+     {________,________,________,________,________,________,________,________,________,________,___XX___,___XX___,___XX___,___XX___,___XX___}
+   },
+
+// Light Control
+[_LYLT] =
+  {
+     {________,________,________,________,________,________,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___,___XX___},
+     {KC_LYDEF,________,________,________,________,________,___XX___,___XX___,___XX___,RGB_HUD ,RGB_HUI ,RGB_SAD ,RGB_SAI ,RGB_VAD ,RGB_VAI },
+     {________,________,________,________,________,________,___XX___,___XX___,___XX___,RGB_TOG ,RGB_RMOD,RGB_MOD ,___XX___,___XX___,___XX___},
+     {________,________,________,________,________,________,___XX___,___XX___,___XX___,BL_TOGG ,BL_DEC  ,BL_INC  ,BL_BRTG ,___XX___,___XX___},
+     {________,________,________,________,________,________,________,________,________,________,___XX___,___XX___,___XX___,___XX___,___XX___}
+  }
+};
+
+
+#define rgblight_setrgb_user_base()  rgblight_show_solid_color(0XED, 0x02, 0x8C)
+#define rgblight_setrgb_user_LYFK()  rgblight_show_solid_color(0xFF, 0x00, 0x00)
+#define rgblight_setrgb_user_LYMED() rgblight_show_solid_color(0x00, 0x00, 0xFF)
+#define rgblight_setrgb_user_LYNUM() rgblight_show_solid_color(0x47, 0x6E, 0x6A)
+#define rgblight_setrgb_user_LYNAV() rgblight_show_solid_color(0x00, 0xFF, 0xFF)
+#define rgblight_setrgb_user_LYMOS() rgblight_show_solid_color(0xFF, 0x8C, 0x00)
+#define rgblight_setrgb_user_LYSYS() rgblight_show_solid_color(0x00, 0xFF, 0x00)
+#define rgblight_setrgb_user_LYLT()  rgblight_show_solid_color(0xFF, 0xFF, 0x00)
+#define rgblight_setrgb_user_LYMD()  rgblight_show_solid_color(0xFF, 0xFF, 0xFF)
+
+//Set a color based on the layer
+uint32_t layer_state_set_user(uint32_t state) {
```

Comment:
I'm guessing that you are having issues with it turning the lights on for the default layer? 

If so, then you need to use `matrix_init_user` to initially set that layer. 

And that's the only place that you'd need the "show solid color" version... or even just `rgblight_enable();`.

Also, in case it helps: https://github.com/qmk/qmk_firmware/blob/master/quantum/rgblight_list.h

Reply:
Thank you for the feedback! The reason I am using the solid color option is that the lighting layer allows you to change modes which is intentionally cleared on layer change. I could do that with a set mode at the top of layer_state_set_user but this method seemed more flexible in case I want one layer to do something special in the future. Similarly, I saw the shortcut list but wanted to tweak a couple shades and liked that the direct color list was right in the keymap. 

I am indeed having issues where the light is off when I first plug in which is fixed the moment I change a layer. I had the code:

//initialize rgb
void matrix_init_user(void) {
  rgblight_setrgb_user_base();
}

just before the layer_set_user yet it seems to make no difference and thus I removed it when I cleaned this up for the PR. It isn't really a big deal to me so just hit a layer key after plugging it in.

Reply:
Well, rgblight_setrgb will force the mode to 1 (solid) anyways.  

In that case, it may be worth trying `rgblight_sethsv(x,x,x)`, and `rgblight_mode(1)` instead.  That may work better.

Reply:
That did it! Thank you very much for the help. I switched all my colors to HSV and now the init works and colors are there on startup.

Reply:
Glad to hear it. :)